packages/typespec-ts/test/commands/runCommand.js

```diff
@@ -1,7 +1,7 @@
 import { spawn } from "child_process";
 
 export async function runCommand(command, args = [], workingDirectory, logger) {
-  return new Promise((resolve, reject) => {
+  return await new Promise((resolve, reject) => {
```

Comment:
There are a few things we can do in this scenario, in order of most to least preferred:

1. `return await` or otherwise await the promise before returning
2. Return the promise and keep the function as async. The runtime [won't wrap the promise in another promise.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function#rewriting_a_promise_chain_with_an_async_function)
3. Change `runCommand` to a synchronous function (i.e. remove `async` from line 3). This is identical to 2, but I prefer the style of 2 over this. `typescript-eslint` [favors 1 over 2 and 3](https://typescript-eslint.io/rules/return-await/) for error handling and perf reasons.

When I wrote this, I didn't know that the runtime only implicitly wraps non-promise values when they're returned from async functions, so I figured that there were potential data race issues. I'd suggest we take on the [`no-floating-promises` typescript-eslint rule](https://typescript-eslint.io/rules/no-floating-promises/) since failing to await a promise is so rarely correct.